src/story_protocol_python_sdk/resources/Permission.py

```diff
@@ -24,75 +27,182 @@ def __init__(self, web3: Web3, account, chain_id: int):
+        :param tx_options dict: [Optional] The transaction options.
+        :return dict: A dictionary with the transaction hash and success status if waiting for transaction.
+        """
+        try:
+            validate_address(signer)
+            
+            self._check_is_registered(ip_id)
+            
+            data = self.access_controller_client.contract.encode_abi(
+                abi_element_identifier="setAllPermissions",
+                args=[
+                    self.web3.to_checksum_address(ip_id),
+                    self.web3.to_checksum_address(signer),
+                    permission
+                ]
+            )
+            
+            response = self.ip_account.execute(
+                to=self.access_controller_client.contract.address,
+                value=0,
+                ip_id=ip_id,
+                data=data,
+                tx_options=tx_options
+            )
+            
+            return {
+                'tx_hash': response['tx_hash']
+            }
+
+        except Exception as e:
+            raise Exception(f"Failed to set all permissions for IP {ip_id} and signer {signer}: {str(e)}")
+
+    def create_set_permission_signature(self, ip_id: str, signer: str, to: str, permission: int, 
+                                        func: str = DEFAULT_FUNCTION_SELECTOR, deadline: int = None, 
+                                        tx_options: dict = None) -> dict:
+        """
+        Specific permission overrides wildcard permission with signature.
+
+        :param ip_id str: The IP ID of the IP account that grants the permission.
+        :param signer str: The address that can call `to` on behalf of the `ip_id`.
+        :param to str: The address that can be called by the `signer`.
+        :param permission int: The new permission level.
+        :param func str: [Optional] The function selector string.
+        :param deadline int: [Optional] The deadline for the signature validity.
+        :param tx_options dict: [Optional] The transaction options.
+        :return dict: A dictionary with the transaction hash and success status if waiting for transaction.
+        """
+        try:
+            validate_address(signer)
+            validate_address(to)
+            
+            self._check_is_registered(ip_id)
+            
+            ip_account_client = IPAccountImplClient(self.web3, contract_address=ip_id)
+            
+            # Convert addresses to checksum format
+            ip_id = self.web3.to_checksum_address(ip_id)
+            signer = self.web3.to_checksum_address(signer)
+            to = self.web3.to_checksum_address(to)
+            
+            data = self.access_controller_client.contract.encode_abi(
+                abi_element_identifier="setTransientPermission",
+                args=[
+                    ip_id,
+                    signer,
+                    to,
+                    Web3.keccak(text=func)[:4] if func else b'\x00\x00\x00\x00',
```

Comment:
can `b'\x00\x00\x00\x00'` be replace with `bytes(4)`?

Reply:
nit: i saw this is used a few times here, maybe we can extract this function selector out? something like `get_selector`?